cmd/soroban-cli/src/commands/contract/extend.rs

```diff
@@ -222,9 +222,12 @@ impl NetworkRunnable for Cmd {
             let entry = client.get_full_ledger_entries(&keys).await?;
             let extension = entry.entries[0].live_until_ledger_seq;
 
-            if entry.latest_ledger + i64::from(extend_to) < i64::from(extension) {
+            let new_ext = entry.latest_ledger + i64::from(extend_to);
+            if new_ext < i64::from(extension) {
                 return Ok(TxnResult::Res(extension));
             }
+
```

Comment:
In the cookbook integration tests, I was seeing situation where we getting here - where `changes` is empty, but the fn does not return. This was then panicking below when trying to get the 0 or 1 index of an empty collection: `match (&changes[0], &changes[1])`.

I'm still not entirely sure why we weren't seeing this issue before - I wonder if it has something to do with the ledger entries being extended in other test scenarios, and therefore we don't end up in the ` if new_ext < i64::from(extension) {` branch? 🤔